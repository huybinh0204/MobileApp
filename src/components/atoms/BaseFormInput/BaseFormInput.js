import { useTheme } from '@emotion/react';
import PropTypes from 'prop-types';
import React, { memo } from 'react';
import { TouchableWithoutFeedback } from 'react-native';
import IconSvg from '_components/atoms/IconSvg/IconSvg';
import { normalize } from '_utilities/screen';
import {
  IconLeftContainer,
  IconRightContainer,
  InputContainer,
  TextInput,
} from './BaseFormInput.styles';

const BaseFormInput = memo(({ actionLeft, actionRight, iconLeft, iconRight, ...rest }) => {
  const { colors } = useTheme();

  return (
    <InputContainer>
      {iconLeft ? (
        <TouchableWithoutFeedback onPress={actionLeft}>
          <IconLeftContainer>
            <IconSvg icon={iconLeft} width={normalize(32)} height={normalize(32)} />
          </IconLeftContainer>
        </TouchableWithoutFeedback>
      ) : null}
      <TextInput
        paddingLeft={!!iconLeft}
        paddingRight={!!iconRight}
        placeholderTextColor={colors.beta100}
        allowFontScaling={false}
        {...rest}
      />
      {iconRight ? (
        <TouchableWithoutFeedback onPress={actionRight}>
          <IconRightContainer>
            <IconSvg icon={iconRight} width={normalize(32)} height={normalize(32)} />
          </IconRightContainer>
        </TouchableWithoutFeedback>
      ) : null}
    </InputContainer>
  );
});

BaseFormInput.defaultProps = {
  actionLeft: () => null,
  actionRight: () => null,
  iconLeft: null,
  iconRight: null,
};

BaseFormInput.propTypes = {
  actionLeft: PropTypes.func,
  actionRight: PropTypes.func,
  iconLeft: PropTypes.string,
  iconRight: PropTypes.string,
};

export default BaseFormInput;
