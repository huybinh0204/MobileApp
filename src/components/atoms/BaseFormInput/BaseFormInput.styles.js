import styled from '@emotion/native';

export const InputContainer = styled.View(({ theme }) => ({
  alignItems: 'center',
  flexDirection: 'row',
  width: '100%',
  borderBottomWidth: 1,
  borderBottomColor: theme.colors.alpha500,
}));

export const TextInput = styled.TextInput(({ theme, paddingLeft, paddingRight }) => ({
  ...theme.typography.mainContentRegular,
  flex: 1,
  paddingLeft: paddingLeft ? theme.spacing.s : theme.spacing.xs,
  paddingRight: paddingRight ? theme.spacing.s : theme.spacing.xs,
  paddingVertical: theme.spacing.s,
}));

export const IconLeftContainer = styled.View(({ theme }) => ({
  paddingLeft: theme.spacing.xs,
}));

export const IconRightContainer = styled.View(({ theme }) => ({
  paddingRight: theme.spacing.xs,
}));
