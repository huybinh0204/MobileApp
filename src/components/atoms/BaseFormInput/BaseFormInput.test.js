import { render } from '@testing-library/react-native';
import React from 'react';
import { ICONS } from '_constants';
import Theme from '_utilities/Theme';
import BaseFormInput from './BaseFormInput';

describe('BaseFormInput', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <Theme>
        <BaseFormInput iconLeft={ICONS.userCircled} value="test" />
      </Theme>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
