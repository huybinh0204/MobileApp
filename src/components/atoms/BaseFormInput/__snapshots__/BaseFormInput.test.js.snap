// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormInput should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#3D3397",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "paddingLeft": 14.5,
        },
        undefined,
      ]
    }
  >
    <SvgMock
      height={58}
      width={58}
    />
  </View>
  <TextInput
    allowFontScaling={false}
    paddingLeft={true}
    paddingRight={false}
    placeholderTextColor="#D4D7DB"
    style={
      Array [
        Object {
          "flex": 1,
          "fontFamily": "Poppins",
          "fontSize": 29,
          "lineHeight": 40,
          "paddingLeft": 29,
          "paddingRight": 14.5,
          "paddingVertical": 29,
        },
        undefined,
      ]
    }
    value="test"
  />
</View>
`;
